Sources/AsyncHTTPClient/HTTPClient.swift

```diff
@@ -590,7 +509,15 @@ public class HTTPClient {
                 connection.release(closing: true, logger: logger)
                 return channel.eventLoop.makeFailedFuture(error)
             }
-        }.always { _ in
+        }
+        .and(task.futureResult)
+        .always { result in
+            if case let .success((_, response)) = result, let httpResponse = response as? HTTPClient.Response {
+                span.attributes.http.statusCode = Int(httpResponse.status.code)
```

Comment:
other that the http attributes it should set Span Status depending on the httpResponse.status.code,

OTel provides a table hot to map one to the other: https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/http.md#status